platform/linux-generic/pktio/socket_xdp.c

```diff
@@ -289,6 +292,13 @@ static int sock_xdp_stats(pktio_entry_t *pktio_entry, odp_pktio_stats_t *stats)
 		stats->in_errors += qi_stats.errors;
 		stats->out_octets += qo_stats.octets;
 		stats->out_packets += qo_stats.packets;
+
+		if (!getsockopt(xsk_socket__fd(sock->xsk), SOL_XDP, XDP_STATISTICS, &xdp_stats,
+				&optlen)) {
+			stats->in_errors += xdp_stats.rx_dropped;
+			stats->in_discards += xdp_stats.rx_invalid_descs;
```

Comment:
Was there a reason for not using `rx_ring_full`? Same applies to `sock_xdp_pktin_queue_stats()`.

Reply:
There seems to be several versions of the statistic structure, `xdp-tools` does not install the header where it is defined. In my environment only dropped rx and invalid descriptor counts are present. In the older version though, ring-full drops are added to `rx_dropped` by kernel.

Reply:
Okay, then it's safer to use only these fields-